bazel/envoy_library.bzl

```diff
@@ -66,6 +66,62 @@ EXTENSION_SECURITY_POSTURES = [
     "data_plane_agnostic",
 ]
 
+# Extension categories as defined by factories
+EXTENSION_CATEGORIES = [
```

Comment:
ive added the list i grepped from the factories in the source code.

not all of these have any coresponding extensions marked. I  guess, at least in some cases, this is because the extensions need additional categories to be marked.

Reply:
also it quite a long list, so not sure if we want to move it to a file and/or have it generated/validated in ci